src/js/game/Entities/Agent.js

```diff
@@ -25,7 +25,9 @@ module.exports = class Agent extends BaseEntity {
         result.plane = "groundPlane";
       }
     } else {
-      if (toPlaceBlock.isRedstone || toPlaceBlock.isRail) {
+      if (toPlaceBlock.isRedstone ||
+      toPlaceBlock.isRail ||
+      toPlaceBlock.blockType === "pressurePlateUp") {
```

Comment:
Why do we also need this change?  Shouldn't the `getIsPlaceableInLiquid` fix be enough?

Reply:
getIsPlaceableInLiquid() change is sufficient to eliminate the ability to place the pressurePlate in the ground plane, which drops into that else statement, but the subsequent if, since it's not redstone or a rail, fails and without the additional inclusion of the pressurePlateUp blocktype check the Agent won't place anything at all on the actionPlane when over ground.

I am like 99% that the spec is that the Agent should be able to place pressurePlates, just like redstone torches, over ground as a source of power for redstoneWire.

Reply:
A better change may be, perhaps, a new getIsPlaceableOverLand() or some such?

Thoughts?

Reply:
Alternatively, we might want to just use `isFlat` for agent, since the only functional restriction on what it should be able to place is stuff that doesn't require a "hop off" animation

Reply:
That's a good suggestion, but I wondered if that might allow for the placement of too many blocks? Are there -no- other blocks other than redstone wire / pressure plates/ rails that would be included in isFlat?

I mean I guess it should be able to place regular torches if it can place redstone torches? What about stuff like flowers? Those are always walkable, right? Are they "flat"?

Reply:
[not flat, no](https://github.com/code-dot-org/craft/blob/master/src/js/game/LevelMVC/LevelBlock.js#L325-L329)